packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -170,110 +157,107 @@ export default function parse(source: string, state: State): TemplateParseResult
-                validateAttributes(element);
-                validateProperties(element);
-
-                parent = stack[stack.length - 1];
-            },
-        },
-
-        Text: {
-            enter(node: parse5.AST.Default.TextNode) {
-                // Extract the raw source to avoid HTML entity decoding done by parse5
-                // TODO [#1286]: Update parse5-with-error to match version used for jsdom (interface for ElementLocation changed)
-                const location = node.__location as parse5.MarkupData.Location;
-
-                const { startOffset, endOffset } = location;
-                const rawText = cleanTextNode(source.slice(startOffset, endOffset));
-
-                if (!rawText.trim().length) {
-                    return;
+        if (children !== undefined) {
+            for (const child of children) {
+                if (treeAdapter.isElementNode(child)) {
+                    const elmNode = parseElement(child, parent);
+                    parsedChildren.push(elmNode);
+                } else if (treeAdapter.isTextNode(child)) {
+                    const textNodes = parseText(child, parent);
+                    parsedChildren.push(...textNodes);
+                } else if (treeAdapter.isCommentNode(child) && preserveComments) {
+                    const commentNode = parseComment(child, parent);
+                    parsedChildren.push(commentNode);
                 }
+            }
+        }
 
-                // Split the text node content arround expression and create node for each
-                const tokenizedContent = rawText.split(EXPRESSION_RE);
+        parent.children = parsedChildren;
+    }
 
-                for (const token of tokenizedContent) {
-                    // Don't create nodes for emtpy strings
-                    if (!token.length) {
-                        continue;
-                    }
+    function parseText(node: parse5.AST.Default.TextNode, parent: IRElement): IRText[] {
+        const parsedTextNodes: IRText[] = [];
 
-                    let value;
-                    if (isExpression(token)) {
-                        try {
-                            value = parseExpression(token, state);
-                        } catch (error) {
-                            addDiagnostic(
-                                normalizeToDiagnostic(
-                                    ParserDiagnostics.TEMPLATE_EXPRESSION_PARSING_ERROR,
-                                    error,
-                                    {
-                                        location: normalizeLocation(location),
-                                    }
-                                )
-                            );
-                            return;
-                        }
-                    } else {
-                        value = decodeTextContent(token);
-                    }
+        // Extract the raw source to avoid HTML entity decoding done by parse5
```

Comment:
I'm confused about this. What would be broken if we didn't do the `source.slice()`?

Reply:
parse5 automatically converts the HTML encoded characters to UTF-8. Which means that at this point we can't differentiate between:

```html
<template>
  <p>&#123; foo &#125;</p>
</template>
```

and 

```html
<template>
  <p>{foo}</p>
</template>
```

We decided early on that if `{` or `}` are HTML encoded those characters shouldn't be treated as expressions delimiters in the template.